addons/stock_account/models/product.py

```diff
@@ -192,8 +192,10 @@ def _prepare_out_svl_vals(self, quantity, company):
             fifo_vals = self._run_fifo(abs(quantity), company)
             vals['remaining_qty'] = fifo_vals.get('remaining_qty')
             # In case of AVCO, fix rounding issue of standard price when needed.
-            if self.product_tmpl_id.cost_method == 'average':
-                rounding_error = currency.round(self.standard_price * self.quantity_svl - self.value_svl)
+            if self.product_tmpl_id.cost_method == 'average' and not float_is_zero(self.quantity_svl, precision_rounding=self.uom_id.rounding):
+                rounding_error = currency.round(
+                    (self.standard_price * self.quantity_svl - self.value_svl) * abs(quantity) / self.quantity_svl
```

Comment:
```suggestion
                    (self.standard_price * self.quantity_svl - self.value_svl) * abs(quantity / self.quantity_svl)
```
The goal is to multiply by the quantity ratio. I would remove the sign on the whoel ratio to be sure